src/drivers/driver.ts

```diff
@@ -603,7 +607,42 @@ export abstract class CMakeDriver implements vscode.Disposable {
       // Expand all important paths
       await this._refreshExpansions();
 
+      const timeStart: number = new Date().getTime();
       const retc = await this.doConfigure(expanded_flags, consumer);
+      const timeEnd: number = new Date().getTime();
+
+      const cmakeVersion = this.cmake.version;
+      const telemetryProperties: {[key: string]: string} = {
+        CMakeExecutableVersion: cmakeVersion ? `${cmakeVersion.major}.${cmakeVersion.minor}.${cmakeVersion.patch}` : '',
+        CMakeGenerator: this.generatorName || '',
+        ConfigType: this.isMultiConf ? 'MultiConf' : this.currentBuildType || '',
+      };
+      const azureSphereTargetApiSet = this.cmakeCacheEntries.get('AZURE_SPHERE_TARGET_API_SET');
+      if (azureSphereTargetApiSet) {
+        telemetryProperties['AzureSphereTargetApiSet'] = azureSphereTargetApiSet.value;
+      }
+      const telemetryMeasures: {[key: string]: number} = {
+        Duration: timeEnd - timeStart,
+      };
+      if (consumer instanceof CMakeOutputConsumer) {
+        let errorCount: number = 0;
+        let warningCount: number = 0;
+        consumer.diagnostics.forEach(v => {
+          if (v.diag.severity === 0) {
+            errorCount++;
+          } else if (v.diag.severity === 1) {
+            warningCount++;
+          }
+        });
+        telemetryMeasures['ErrorCount'] = errorCount;
+        telemetryMeasures['WarningCount'] = warningCount;
+      } else {
+        // Wrong type: shouldn't get here, just in case
+        rollbar.error('Wrong build result type.');
+        telemetryMeasures['ErrorCount'] = retc ? 1 : 0;
+      }
+      telemetry.logEvent('parse', telemetryProperties, telemetryMeasures);
```

Comment:
This event should probably be called "configure"

Reply:
Sure. That's the named used in VS but we can't really be consistent with them anyway...